### ISSUE TITLE
leaf_node.hpp style issues

### ISSUE THREAD

**Issue by nighelles** [rethinkdb/rethinkdb#5288]
```
../src/btree/leaf_node.hpp:11:  Found Boost header after other header. Should be: leaf_node.h, c system, c++ system, other.  [build/include_order] [4]
../src/btree/leaf_node.hpp:13:  "arch/compiler.hpp" already included at ../src/btree/leaf_node.hpp:9  [build/include] [4]
```

**Comment by danielmewes** [comment]
@AtnNn We assigned this to you because at least the second line looks related to the Windows port.

**Comment by AtnNn** [comment]
fixed in 70cd8d0
